### PR TITLE
Fixes #88: Filter out "#pragma once in main file" from header files on C/C++

### LINKED ISSUE #88: #pragma once in main file
I'm getting `#pragma once in main file` when viewing any header file with `#pragma once`. Is there a way to disable this? (clang v3.4)

### PR DESCRIPTION
**Pull request by lbguilherme**
Fixes #88.

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -60,18 +60,24 @@ export default {
...
           "-fexceptions"];
 
+        let filterRegex = null;
         const grammar = activeEditor.getGrammar().name;
 
         if(/^C\+\+/.test(grammar)) {
           //const language = "c++";
           args.push("-xc++");
           args.push(...atom.config.get("linter-clang.clangDefaultCppFlags").split(/\s+/));
+
```
Comment (Arcanemagus) [inline lib/main.js]: You have blank whitespace on this line.
image

File: lib/main.js [hunk 2]
```diff
@@ -82,6 +88,10 @@ export default {
           //const language = "c";
           args.push("-xc");
           args.push(...atom.config.get("linter-clang.clangDefaultCFlags").split(/\s+/));
+
```
Comment (Arcanemagus) [inline lib/main.js]: More whitespace.

File: lib/main.js (hunk 1 above)
Comment (Arcanemagus) [inline lib/main.js]: Just to be clear, I'm not saying you need to remove the line itself, just the unnecessary whitespace :wink:.

**Comment by lbguilherme** [comment]
Oops, bad editor settings.

Fixed and rebased.

**Comment by Arcanemagus** [comment]
@kepler0 Do you have a setup to test this with? Looks like it should work from the code but I'll leave merging to you.

**Comment by kepler0** [comment]
@Arcanemagus The unit tests should indicate if this introduce a regression, but I'll check for this specific code with some iOS code I have lying around.

**Comment by kepler0** [comment]
Ignore the AppVeyor results those have not yet worked for this project.

**Comment by kepler0** [comment]
Ignore Circle CI. The builds do not seem to be working there anymore. But now that we can unit test Atom on Linux with Travis I may just remove Circle CI support.

**Comment by kepler0** [comment]
@lbguilherme Could you merge in or rebase the latest changes from master. That will allow better CI testing to verify the results.

**Comment by kepler0** [comment]
As well, if you could create unit tests which test header files that report the pragma warning and assert that a message that contains the warning isn't returned that would be great.

**Comment by lbguilherme** [comment]
How can I run the specs locally?

**Comment by lbguilherme** [comment]
I'm not really sure what is wrong on the tests, this works manually testing on atom :/

**Comment by kepler0** [comment]
That's interesting. I will investigate. Thanks for the unit tests, makes it much easier to test.

**Comment by kepler0** [comment]
Tests failing locally.

**Comment by kepler0** [comment]
Failing because it's not filtering out for me on OS X 10.11.

I added some `console-log` lines to debug the results being given and it does have a linter message object:
```js
{
  type: 'warning',
  text: '#pragma once in main file'
}
```

**Comment by hammackj** [comment]
Any update on this?

**Comment by Arcanemagus** [comment]
I've re-implemented parts of this in https://github.com/AtomLinter/linter-clang/pull/191/commits/36f4ac5f398f46be6c4b03282f23ba058a778467, replacing the filtering logic with the use of `-Wno-pragma-once-outside-header` as suggested by @jplatte in https://github.com/AtomLinter/linter-clang/issues/88#issuecomment-219218240.

Thanks for taking the time to work on this @lbguilherme and I'm sorry this wasn't dealt with sooner!
